runtime-transaction/src/instruction_details.rs

```diff
@@ -0,0 +1,361 @@
+use {
+    crate::{builtin_instruction_details::*, compute_budget_instruction_details::*},
+    solana_compute_budget::compute_budget_limits::*,
+    solana_sdk::{
+        instruction::{CompiledInstruction, InstructionError},
+        pubkey::Pubkey,
+        transaction::{Result, TransactionError},
+    },
+    std::num::NonZeroU32,
+};
+
+/// Information about instructions gathered after scan over transaction;
+/// These are "raw" information that suitable for cache and reuse.
+#[cfg_attr(test, derive(Eq, PartialEq))]
+#[derive(Default, Debug)]
+pub struct InstructionDetails {
+    builtin_instruction_details: BuiltinInstructionDetails,
+    compute_budget_instruction_details: ComputeBudgetInstructionDetails,
+}
```

Comment:
I don't think I see the value of combining different "instruction details" into a single struct. Can we keep them separate? I don't see much value in eliminating a loop since the work inside the loop is more expensive than the loop itself.

Reply:
I kind like this umbrella struct, it defines concept well: giving the list of instructions, it fetches and stores their "details". What "details" are can just be internal knowledge. Information in `builtin_instruction_details` and `compute_budget_instruction_details` are used together (in `num_non_compute_budget_instructions()`), they probably could be flatted into one `InstructionDetails` struct. From call site perspective, it's easier to work with one `instructionDetails` vs different "details".

Reply:
787326d711edf42d22fe71efcc517bd90fa68dfb removed the "inner details" structs.

Reply:
I prefer to keep the inner structs and remove the outer struct

Reply:
What I tried to say was the "inner details" are not truly independent two set of datas, rather they were artificially separated.

Can you help me to understand why keeping inner structs is better than what it is?